### DISCUSSION TITLE
PageState.ModuleId -1 causing issues with API Authorization

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#1906] [General]
I'm using the `[Authorize(Policy = PolicyNames.ViewModule)]` attribute on my API controller methods and I noticed when I make controller calls from my customer Razor components sometimes this value is -1. That causes the controller method to not get called.  The way I'm passing my PageState is via a CascadingParameter that is set in app.razor as a `<CascadingValue Value="@PageState">`.

My current work-around is to remove the authorization.  Is there another way I should be accessing the pagestate?  I could pass it to all my custom components, that that seems like the same as a cascading parameter so I assume the results would be the same.

**Comment by leigh-pointer** [comment]
your component should inherit from ModuleBase

**Reply by markdav-is** [reply]
It's not a component page... but a dropdown list.  It's smart enough to get it's own data, so it needs to make API calls.  Do you think ModuleBase is appropriate for a shared component that can be used in many modules?

**Reply by leigh-pointer** [reply]
@markdav-is  Being a control you should use 

`@inherits ModuleControlBase`

My rule is to always look at the framework and follow the rule there.  If you look at the Client Controls you see these controls inherit from ModuleControlBase

I hope this helps

**Reply by markdav-is** [reply]
thanks so much!  I'm still seeing a -1 from time to time when I'm processing events.  Also thanks for the tip on checking in the framework client controls.  I need to be researching there first and gaining a fuller understanding of the framework.

**Reply by markdav-is** [reply]
@inherits ModuleControlBase has no effect on the -1 showing up.  Maybe it's a control lifecycle thing and I'm checking it too early?  I'll keep digging and share the answer when I find it.
